File: 프로그래머스/0/120811. 중앙값 구하기/중앙값 구하기.js
function solution(array) {
    var answer = 0;
    var len = array.length;
    var middleIndex = Math.floor(len / 2); // 중간 인덱스 계산

    // 배열을 크기대로 정렬
    array.sort(function(a, b) {
        return a - b;
    });

    // 배열의 중간 값을 반환
    if (len % 2 === 1) { // 홀수 길이인 경우
        answer = array[middleIndex];
    } else { // 짝수 길이인 경우
        // 중간 두 값의 평균 반환
        answer = (array[middleIndex - 1] + array[middleIndex]) / 2;
    }

    return answer;
}

File: 프로그래머스/0/120811. 중앙값 구하기/중앙값 구하기.java
import java.util.Arrays;

class Solution {
    public double solution(int[] array) {
        Arrays.sort(array);
        int length = array.length;
        if (length % 2 == 0) {
            // 배열의 길이가 짝수인 경우 중앙의 두 수의 평균을 반환합니다.
            return (array[length / 2 - 1] + array[length / 2]) / 2.0;
        } else {
            // 배열의 길이가 홀수인 경우 중앙의 수를 반환합니다.
            return array[length / 2];
        }
    }
}